private/bufpkg/bufimage/bufimagebuild/bufimagebuild.go

```diff
@@ -52,6 +53,36 @@ type BuildOption func(*buildOptions)
 // WithExcludeSourceCodeInfo returns a BuildOption that excludes sourceCodeInfo.
 func WithExcludeSourceCodeInfo() BuildOption {
 	return func(buildOptions *buildOptions) {
-		buildOptions.excludeSourceCodeInfo = true
+		buildOptions.sourceInfoMode = protocompile.SourceInfoNone
+	}
+}
+
+// WithExtraSourceCodeInfo returns a BuildOption that causes source code info to
+// be generated with extra comments.
+func WithExtraSourceCodeInfo() BuildOption {
+	return func(buildOptions *buildOptions) {
+		buildOptions.sourceInfoMode = protocompile.SourceInfoExtraComments
+	}
+}
+
+// WithParallelism indicates the maximum level of parallelism to use when building
+// an image. If unspecified, the maximum parallelism will be automatically set to
+// min(runtime.NumCPU(), runtime.GOMAXPROCS(-1)).
+func WithParallelism(parallelism int) BuildOption {
```

Comment:
Is this different than before? All parallelism in bufbuild/buf should be controlled by `pkg/thread`, so whatever `pkg/thread` has set for parallelism should be what we use within `bufimagebuild`.

Reply:
I'll rework it. I originally added this before discovering the `thread` package and seeing how it was used before. That is in fact how we set the _default_ parallelism. The only caller of this is a test (that sets it to 1 for deterministic output). I can instead have the test set the parallelism via the `thread` package (and defer to restore to prior setting).

Reply:
Removed this option.